Week3/mysql-connection.js

```diff
@@ -0,0 +1,46 @@
+import mysql from 'mysql';
+import colors from 'colors';
+import util from 'util';
+/**
+ * Make a connection to your database, using your MySQL
+ * hyfuser login credentials
+ * */
+export const connection = mysql.createConnection({
+  host: 'localhost',
+  user: 'hyfuser',
+  password: 'hyfpassword',
+  charset: 'utf8',
+});
+
+export const makeQuery = util.promisify(connection.query.bind(connection));
+
+export function makeQueries(queriesArray) {
+  return Promise.all(queriesArray.map((q) => makeQuery(q)));
+}
+
+export const insertArray = (array, query) => Promise.all(array.map((r) => makeQuery(query, [r])));
```

Comment:
At first this function seemed to be doing the exact same thing as `makeQueries`.
But after reading it carefully it became clear that it's slightly different, it uses array map on the `array` instead of the `queries`

The advantage of splitting your code into files like this is that you can make functions 'bigger' instead of one-linners.
And improve the readability in the process :)

Using other variables would have helped, for example like this:

```suggestion
export function insertArray(values, insertQuery)  {
 return Promise.all(values.map((value) => makeQuery(insertQuery, [value])));
}
```